OSXDemo/OCTFriendsViewController.m

```diff
@@ -0,0 +1,307 @@
+    NSIndexSet *columnSet = [[NSIndexSet alloc] initWithIndex:0];
+
+
+    switch (type) {
+        case RBQFetchedResultsChangeInsert:
+            [tableView insertRowsAtIndexes:newSet withAnimation:NSTableViewAnimationSlideRight];
+            break;
+        case RBQFetchedResultsChangeDelete:
+            [tableView removeRowsAtIndexes:oldSet withAnimation:NSTableViewAnimationSlideLeft];
+            break;
+        case RBQFetchedResultsChangeUpdate:
+            [tableView reloadDataForRowIndexes:oldSet columnIndexes:columnSet];
+            break;
+        case RBQFetchedResultsChangeMove:
+            [tableView removeRowsAtIndexes:oldSet withAnimation:NSTableViewAnimationSlideLeft];
+            [tableView insertRowsAtIndexes:newSet withAnimation:NSTableViewAnimationSlideRight];
+            break;
+    }
+}
+
+- (void)controllerDidChangeContent:(RBQFetchedResultsController *)controller
+{
+    NSTableView *tableView = (controller == self.friendRequestResultsController) ? self.requestsTableView : self.friendsTableView;
+
+    @try {
+        [tableView endUpdates];
+    }
+    @catch (NSException *ex) {
+        [controller reset];
+        [tableView reloadData];
+    }
+}
+
+#pragma mark - Actions
+
+- (IBAction)addFriendReturn:(NSTextField *)sender
+{
+    [self.manager.friends
+     sendFriendRequestToAddress:sender.stringValue
+                        message:@"Friend request from objcTox Mac OSX Demo"
+                          error:nil];
+}
+
+- (IBAction)removeFriendButtonPressed:(NSButton *)sender
+{
+    NSInteger selectedRow = self.friendsTableView.selectedRow;
+
+    if (selectedRow == -1) {
+        return;
+    }
+
+    NSIndexPath *path = [NSIndexPath indexPathForRow:selectedRow inSection:0];
+    OCTFriend *friend = [self.friendResultsController objectAtIndexPath:path];
+
+    [self.manager.friends removeFriend:friend error:nil];
+}
+
+#pragma mark - Private
+
+- (void)acceptFriendRequests
+{
+    NSInteger selectedRow = self.requestsTableView.selectedRow;
+
+    if (selectedRow == -1) {
+        return;
+    }
+
+    NSIndexPath *path = [NSIndexPath indexPathForRow:selectedRow inSection:0];
+    OCTFriendRequest *request = [self.friendRequestResultsController objectAtIndexPath:path];
+
+    [self.manager.friends approveFriendRequest:request error:nil];
+
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self.requestsTableView reloadData];
+    });
+}
+
+- (void)rejectFriendRequests
+{
+    NSInteger selectedRow = self.requestsTableView.selectedRow;
+
+    if (selectedRow == -1) {
+        return;
+    }
+
+    NSIndexPath *path = [NSIndexPath indexPathForRow:selectedRow inSection:0];
+    OCTFriendRequest *request = [self.friendRequestResultsController objectAtIndexPath:path];
+
+    [self.manager.friends removeFriendRequest:request];
+
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self.requestsTableView reloadData];
+    });
```

Comment:
Also these two methods (accept & reject friend request) can be refactored to one.

```objc
- (void)proceedWithFriendRequest:(BOOL)accept
```

Or some method with ever better name ;)